metaflow/datatools/s3.py

```diff
@@ -277,7 +280,7 @@ def error(self):
         return self._s3_error
 
     def reset_client(self):
-        self._s3_client, self._s3_error = get_s3_client()
+        self._s3_client, self._s3_error = get_s3_client(role=self._s3_role)
```

Comment:
as a mandatory naming nitpick for this PR, consider calling this `s3_role_arn` to make it clear that you have to pass the ARN :)